src/SFA.DAS.EmployerApprenticeshipsService.Application/Queries/GetAccountEmployerAgreements/GetAccountEmployerAgreementsQueryHandler.cs

```diff
@@ -39,11 +57,38 @@ public GetAccountEmployerAgreementsQueryHandler(IAccountRepository accountReposi
                 throw new UnauthorizedAccessException();
             }
 
-            var agreements = await _accountRepository.GetEmployerAgreementsLinkedToAccount(_hashingService.DecodeValue(message.HashedAccountId));
+            var accountId = _hashingService.DecodeValue(message.HashedAccountId);
 
+            var agreements =
+                await _db.Agreements
+                    .Where(ag => ag.AccountId == accountId
+                                 && (ag.StatusId == EmployerAgreementStatus.Signed ||
+                                     ag.StatusId == EmployerAgreementStatus.Pending))
+                    .GroupBy(grp => grp.LegalEntity)
+                    .ProjectTo<EmployerAgreementStatusDto>(_configurationProvider)
+                    .ToListAsync();
+                                                    
             foreach (var agreement in agreements)
             {
-                agreement.HashedAgreementId = _hashingService.HashValue(agreement.Id);
+                agreement.AccountId = accountId;
+                agreement.HashedAccountId = message.HashedAccountId;
+
+                if (agreement.HasSignedAgreement)
+                {
+                    agreement.Signed.HashedAgreementId = _hashingService.HashValue(agreement.Signed.Id);
+                }
+
+                if (agreement.HasPendingAgreement)
+                {
+                    if (agreement.HasSignedAgreement && agreement.Signed.VersionNumber > agreement.Pending.VersionNumber)
```

Comment:
Is it possible to do the version check be done in the query so that if Rob reuses the Dto he doesn't have to duplicate this logic?